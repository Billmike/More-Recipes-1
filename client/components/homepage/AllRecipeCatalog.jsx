import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import RecipeCatalogCard from '../common/RecipeCatalogCard';

const propTypes = {
  recipesCatalog: PropTypes.arrayOf(PropTypes.shape()),
};

const defaultProps = {
  recipesCatalog: undefined,
};

/**
 * All recipes catalog component
 *
 * @param {object} props - The properties passsed to the component
 *
 * @returns {object} React element
 */
const AllRecipeCatalog = (props) => {
  const { recipesCatalog } = props;

  return (
    <div className="row">
      <div className="col s12" >
        <div className="row valign-wrapper" >
          <h4 className="col s11">All Recipes</h4>
          <span className="col s1 right" >
            <Link to="/catalog">View All &gt;</Link>
          </span>
        </div>
      </div>
      {recipesCatalog && recipesCatalog.map((recipe, index) =>
        (<RecipeCatalogCard
          {...props}
          key={recipe.id}
          index={index}
          recipe={recipe}
        />))}
    </div>
  );
};

AllRecipeCatalog.propTypes = propTypes;
AllRecipeCatalog.defaultProps = defaultProps;

export default AllRecipeCatalog;
