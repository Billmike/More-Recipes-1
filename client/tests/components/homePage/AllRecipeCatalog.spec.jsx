import React from 'react';

import AllRecipeCatalog from '../../../components/homepage/AllRecipeCatalog';

describe('AllRecipeCatalog component', () => {
  it('should render the component', () => {
    const props = {
      recipesCatalog: [],
      recipes: {
        isFetchingRecipes: false,
      },
    };
    const wrapper = shallow(<AllRecipeCatalog {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with spinner when fetching recipe', () => {
    const props = {
      recipesCatalog: [],
      recipes: {
        isFetchingRecipes: true,
      },
    };
    const wrapper = shallow(<AllRecipeCatalog {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('Spinner').length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render component with recipes card', () => {
    const props = {
      recipesCatalog: [{
        id: 1,
        title: 'Rice',
        category: 'Dinner',
        description: 'Tasty rice',
        preparationTime: 69,
        ingredients: 'Rice, tomatoes, salt',
        directions: 'Do this. Do this. Do that.',
        upvotes: 2,
        downvotes: 0,
        favorites: 5,
        votes: 3,
      }, {
        id: 2,
        title: 'Yam',
        category: 'Breakfast',
        description: 'Tasty yam',
        preparationTime: 50,
        ingredients: 'Yam, tomatoes, egg, salt',
        directions: 'Do this. Do this. Do that.',
        upvotes: 5,
        downvotes: 2,
        favorites: 4,
        votes: 5,
      }],
      recipes: {
        isFetchingRecipes: false,
      },
    };
    const wrapper = shallow(<AllRecipeCatalog {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper.find('RecipeCatalogCard').length).toEqual(2);
    expect(wrapper).toMatchSnapshot();
  });
});
