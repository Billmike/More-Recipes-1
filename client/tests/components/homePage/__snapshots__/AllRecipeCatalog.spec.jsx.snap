// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllRecipeCatalog component should render component with recipes card 1`] = `
<div
  className="row"
>
  <div
    className="col s12"
  >
    <div
      className="row valign-wrapper"
    >
      <h4
        className="col s8 m10 l11"
      >
        All Recipes
      </h4>
      <span
        className="col s4 m2 l1  right"
        id="view-catalog"
      >
        <Link
          replace={false}
          to="/catalog"
        >
          View All &gt;
        </Link>
      </span>
    </div>
  </div>
  <div>
    <RecipeCatalogCard
      index={0}
      key="1"
      recipe={
        Object {
          "category": "Dinner",
          "description": "Tasty rice",
          "directions": "Do this. Do this. Do that.",
          "downvotes": 0,
          "favorites": 5,
          "id": 1,
          "ingredients": "Rice, tomatoes, salt",
          "preparationTime": 69,
          "title": "Rice",
          "upvotes": 2,
          "votes": 3,
        }
      }
      recipes={
        Object {
          "isFetchingRecipes": false,
        }
      }
      recipesCatalog={
        Array [
          Object {
            "category": "Dinner",
            "description": "Tasty rice",
            "directions": "Do this. Do this. Do that.",
            "downvotes": 0,
            "favorites": 5,
            "id": 1,
            "ingredients": "Rice, tomatoes, salt",
            "preparationTime": 69,
            "title": "Rice",
            "upvotes": 2,
            "votes": 3,
          },
          Object {
            "category": "Breakfast",
            "description": "Tasty yam",
            "directions": "Do this. Do this. Do that.",
            "downvotes": 2,
            "favorites": 4,
            "id": 2,
            "ingredients": "Yam, tomatoes, egg, salt",
            "preparationTime": 50,
            "title": "Yam",
            "upvotes": 5,
            "votes": 5,
          },
        ]
      }
    />
    <RecipeCatalogCard
      index={1}
      key="2"
      recipe={
        Object {
          "category": "Breakfast",
          "description": "Tasty yam",
          "directions": "Do this. Do this. Do that.",
          "downvotes": 2,
          "favorites": 4,
          "id": 2,
          "ingredients": "Yam, tomatoes, egg, salt",
          "preparationTime": 50,
          "title": "Yam",
          "upvotes": 5,
          "votes": 5,
        }
      }
      recipes={
        Object {
          "isFetchingRecipes": false,
        }
      }
      recipesCatalog={
        Array [
          Object {
            "category": "Dinner",
            "description": "Tasty rice",
            "directions": "Do this. Do this. Do that.",
            "downvotes": 0,
            "favorites": 5,
            "id": 1,
            "ingredients": "Rice, tomatoes, salt",
            "preparationTime": 69,
            "title": "Rice",
            "upvotes": 2,
            "votes": 3,
          },
          Object {
            "category": "Breakfast",
            "description": "Tasty yam",
            "directions": "Do this. Do this. Do that.",
            "downvotes": 2,
            "favorites": 4,
            "id": 2,
            "ingredients": "Yam, tomatoes, egg, salt",
            "preparationTime": 50,
            "title": "Yam",
            "upvotes": 5,
            "votes": 5,
          },
        ]
      }
    />
  </div>
</div>
`;

exports[`AllRecipeCatalog component should render component with spinner when fetching recipe 1`] = `
<div
  className="row"
>
  <div
    className="col s12"
  >
    <div
      className="row valign-wrapper"
    >
      <h4
        className="col s8 m10 l11"
      >
        All Recipes
      </h4>
      <span
        className="col s4 m2 l1  right"
        id="view-catalog"
      >
        <Link
          replace={false}
          to="/catalog"
        >
          View All &gt;
        </Link>
      </span>
    </div>
  </div>
  <div
    className="center-align"
  >
    <Spinner
      size="big"
    />
  </div>
</div>
`;

exports[`AllRecipeCatalog component should render the component 1`] = `
<div
  className="row"
>
  <div
    className="col s12"
  >
    <div
      className="row valign-wrapper"
    >
      <h4
        className="col s8 m10 l11"
      >
        All Recipes
      </h4>
      <span
        className="col s4 m2 l1  right"
        id="view-catalog"
      >
        <Link
          replace={false}
          to="/catalog"
        >
          View All &gt;
        </Link>
      </span>
    </div>
  </div>
  <div />
</div>
`;
